pkg/database/extensions.go

```diff
@@ -0,0 +1,124 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/base64"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/encrypt"
+)
+
+var _ codec.InterfaceExt = (*SecureByteExt)(nil)
+
+type SecureByteExt struct {
+	Log    *logrus.Entry
+	Cipher encrypt.Cipher
+}
+
+func (s SecureByteExt) ConvertExt(v interface{}) interface{} {
+	data := v.(api.SecureByte)
+	var encrypted []byte
+	if s.Cipher != nil {
+		var err error
+		encrypted, err = s.Cipher.Encrypt(data)
```

Comment:
panic, do not log.  The encoder will recover your panic and return an error to the caller.